geomesa-accumulo/geomesa-accumulo-tools/src/main/scala/org/locationtech/geomesa/accumulo/tools/ingest/AccumuloBulkIngestCommand.scala

```diff
@@ -0,0 +1,188 @@
+  override val params = new AccumuloBulkIngestParams()
+
+  override protected def startIngest(
+      mode: RunMode,
+      ds: AccumuloDataStore,
+      sft: SimpleFeatureType,
+      converter: Config,
+      inputs: Inputs): Awaitable = {
+
+    val maxSplitSize =
+      if (!params.combineInputs) { None } else { Option(params.maxSplitSize).map(_.intValue()).orElse(Some(0)) }
+
+    // validate index param now that we have a datastore and the sft has been created
+    val index = params.loadIndex(ds, sft.getTypeName, IndexMode.Write).map(_.identifier)
+
+    val partitions = TablePartition(ds, sft).map { tp =>
+      if (params.cqlFilter == null) {
+        throw new ParameterException(
+          s"Schema '${sft.getTypeName}' is a partitioned store. In order to bulk load, the '--cql' parameter " +
+              "must be used to specify the range of the input data set")
+      }
+      tp.partitions(params.cqlFilter).filter(_.nonEmpty).getOrElse {
+        throw new ParameterException(
+          s"Partition filter does not correspond to partition scheme ${tp.getClass.getSimpleName}. Please specify " +
+              "a valid filter using the '--cql' parameter")
+      }
+    }
+
+
+    mode match {
+      case RunModes.Local =>
+        throw new IllegalArgumentException("Bulk ingest must be run in distributed mode")
+
+      case RunModes.Distributed =>
+        val conf = new Configuration()
+        // file output format doesn't let you write to an existing directory
+        val output = new Path(params.outputPath)
+        val context = FileContext.getFileContext(output.toUri, conf)
+        if (context.util.exists(output)) {
+          val warning = s"Output directory '$output' exists"
+          if (params.force) {
+            Command.user.warn(s"$warning - deleting it")
+          } else if (!Prompt.confirm(s"WARNING DATA MAY BE LOST: $warning. Delete it and continue (y/n)? ")) {
+            throw new ParameterException(s"Output directory '$output' exists")
+          }
+          context.delete(output, true)
+        }
+
+        val tempPath = Option(params.tempPath).map { temp =>
+          val path = new Path(temp)
+          // get a new file context as this is likely to be a different filesystem (i.e. hdfs vs s3)
+          val tempContext = FileContext.getFileContext(path.toUri, conf)
+          val dir = tempContext.makeQualified(path)
+          if (tempContext.util.exists(dir)) {
+            Command.user.info(s"Deleting temp output path $dir")
+            tempContext.delete(dir, true)
+          }
+          dir
+        }
+
+        Command.user.info(s"Running bulk ingestion in distributed ${if (params.combineInputs) "combine " else "" }mode")
+        new BulkConverterIngest(ds, connection, sft, converter, inputs.paths, output, tempPath, maxSplitSize,
+          index, partitions, libjarsFiles, libjarsPaths)
+
+      case _ =>
+        throw new NotImplementedError(s"Missing implementation for mode $mode")
```

Comment:
Do we want to call out DistributedCombine specifically?

Reply:
distributed combine got removed as a standalone option, it's just distributed and local now, and you can specify `--combine-inputs` to get the effect of distributed combine.